render/batchload.go

```diff
@@ -1,152 +1,88 @@
 package render
 
 import (
-	"encoding/json"
-	"errors"
+	"io/fs"
+	"path"
 	"path/filepath"
 	"regexp"
 	"strconv"
-	"strings"
 	"sync"
 
+	"github.com/oakmound/oak/v3/alg/intgeom"
 	"github.com/oakmound/oak/v3/dlog"
 	"github.com/oakmound/oak/v3/fileutil"
 	"github.com/oakmound/oak/v3/oakerr"
 )
 
-var (
-	regexpSingleNumber = regexp.MustCompile(`^\d+$`)
-	regexpTwoNumbers   = regexp.MustCompile(`^\d+x\d+$`)
-)
-
 // BatchLoad loads subdirectories from the given base folder and imports all files,
 // using alias rules to automatically determine the size of sprites and sheets in
 // subfolders.
-// A folder named 16x8 will have its images split into sheets where each sprite is
-// 16x8, for example. 16 is a shorter way of writing 16x16.
-// An alias.json file can be included that can indicate what dimensions named folders
-// represent, so a "tiles": "32" field in the json would indicate that sprite sheets
-// in the /tiles folder should be read as 32x32
 func BatchLoad(baseFolder string) error {
 	return BlankBatchLoad(baseFolder, 0)
 }
 
 // BlankBatchLoad acts like BatchLoad, but will not load and instead return a blank image
 // of the appropriate dimensions for anything above maxFileSize.
 func BlankBatchLoad(baseFolder string, maxFileSize int64) error {
-	folders, err := fileutil.ReadDir(baseFolder)
-	if err != nil {
-		return err
-	}
-	aliases, err := parseAliasFile(baseFolder)
-	if err != nil {
-		return err
-	}
-
-	warnFiles := []string{}
-
 	var wg sync.WaitGroup
-
-	for _, folder := range folders {
-		if folder.IsDir() {
-			frameW, frameH, possibleSheet, err := parseLoadFolderName(aliases, folder.Name())
+	err := fs.WalkDir(fileutil.FS, baseFolder, func(file string, d fs.DirEntry, err error) error {
+		if d == nil {
+			// We've been given a bad base directory
+			return oakerr.InvalidInput{InputName: "baseFolder"}
+		}
+		if d.IsDir() {
+			return nil
+		}
+		wg.Add(1)
+		go func(file string) {
```

Comment:
Thought for v4, consider if we can allow slotting of additional decoders based on regex which would help make this a bit more extensible.

Reply:
What is an example of a file that would be decoded by something other than its extension?